examples/transfer.rs

```diff
@@ -0,0 +1,100 @@
+use std::{path::PathBuf, str::FromStr};
+
+use anyhow::Result;
+use iroh::{protocol::Router, Endpoint};
+use iroh_base::ticket::BlobTicket;
+use iroh_blobs::{
+    net_protocol::Blobs,
+    rpc::client::blobs::{ReadAtLen, WrapOption},
+    store::mem,
+    util::{local_pool::LocalPool, SetTagOption},
+};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Create an endpoint, it allows creating and accepting
+    // connections in the iroh p2p world
+    let endpoint = Endpoint::builder().discovery_n0().bind().await?;
+
+    // We initialize the Blobs protocol in-memory
+    let local_pool = LocalPool::default();
+    let blobs = Blobs::memory().build(&local_pool, &endpoint);
+
+    // Now we build a router that accepts blobs connections & routes them
+    // to the blobs protocol.
+    let node = Router::builder(endpoint)
+        .accept(iroh_blobs::ALPN, blobs)
+        .spawn()
+        .await?;
+
+    let args = std::env::args().collect::<Vec<_>>();
+    match &args.iter().map(String::as_str).collect::<Vec<_>>()[..] {
+        [_cmd, "send", path] => {
+            let abs_path = PathBuf::from_str(path)?.canonicalize()?;
+
+            let blobs = node
+                .get_protocol::<Blobs<mem::Store>>(iroh_blobs::ALPN)
+                .unwrap()
+                .client();
+
+            println!("Analyzing file.");
+
+            let blob = blobs
+                .add_from_path(abs_path, true, SetTagOption::Auto, WrapOption::NoWrap)
+                .await?
+                .finish()
+                .await?;
+
+            let node_id = node.endpoint().node_id();
+            let ticket = BlobTicket::new(node_id.into(), blob.hash, blob.format)?;
+
+            println!("File analyzed. Fetch this file by running:");
+            println!("cargo run --example transfer -- receive {ticket} {path}");
+
+            tokio::signal::ctrl_c().await?;
+        }
+        [_cmd, "receive", ticket, path] => {
+            let path_buf = PathBuf::from_str(path)?;
+            let ticket = BlobTicket::from_str(ticket)?;
+
+            let blobs = node
```

Comment:
I guess you want to show how you can get blobs from the router, but if this was production code I would just keep the blobs from above around.

As I mentioned in https://github.com/n0-computer/iroh/pull/2938 I am not convinced that the downcast stuff is pulling its weight.